website/api/manager/ManagerAPI.php

```diff
@@ -81,20 +88,29 @@ protected function task() {
                 );
             }
 
+            // don't give out any task if games have been stopped.
+            if($this->checkConfig("compState", "noGameTasks")) {
+                return array("type" => "notask");
+            }
+
             // Assign a run game tasks
             $possibleNumPlayers = array(2, 2, 2, 2, 2, 3, 3, 3, 3, 4, 4, 4, 5, 5, 6);
             $numPlayers = $possibleNumPlayers[array_rand($possibleNumPlayers)];
 
             $seedPlayer = null;
-            $randValue = mt_rand() / mt_getrandmax();
-            if($randValue > 0.5) {
-                $seedPlayer = $this->select("SELECT * FROM User WHERE isRunning = 1 and numGames < 400 order by rand()*-pow(sigma, 2) LIMIT 1");
-            }
-            if ($randValue > 0.25 && $randValue <= 0.5) {
-                $seedPlayer = $this->select("SELECT * FROM (SELECT u.* FROM (SELECT MAX(g.timestamp) as maxTime, gu.userID as userID FROM GameUser gu INNER JOIN Game g ON g.gameID=gu.gameID GROUP BY gu.userID) temptable INNER JOIN User u on u.userID = temptable.userID where numGames < 400 and isRunning = 1 order by maxTime ASC limit 15) orderedTable order by rand() limit 1;");
-            } 
-            if($randValue <= 0.25 || count($seedPlayer) < 1) {
-                $seedPlayer = $this->select("SELECT u.* FROM (SELECT MAX(g.timestamp) as maxTime, gu.userID as userID FROM GameUser gu INNER JOIN Game g ON g.gameID=gu.gameID GROUP BY gu.userID) temptable INNER JOIN User u on u.userID = temptable.userID where isRunning = 1 order by maxTime ASC limit 1");
+            if(!$this->checkConfig("compState", "finalsPairing")) {
+                $randValue = mt_rand() / mt_getrandmax();
+                if($randValue > 0.5) {
+                    $seedPlayer = $this->select("SELECT * FROM User WHERE isRunning = 1 and numGames < 400 order by rand()*-pow(sigma, 2) LIMIT 1");
+                }
+                if ($randValue > 0.25 && $randValue <= 0.5) {
+                    $seedPlayer = $this->select("SELECT * FROM (SELECT u.* FROM (SELECT MAX(g.timestamp) as maxTime, gu.userID as userID FROM GameUser gu INNER JOIN Game g ON g.gameID=gu.gameID GROUP BY gu.userID) temptable INNER JOIN User u on u.userID = temptable.userID where numGames < 400 and isRunning = 1 order by maxTime ASC limit 15) orderedTable order by rand() limit 1;");
+                }
+                if($randValue <= 0.25 || count($seedPlayer) < 1) {
+                    $seedPlayer = $this->select("SELECT u.* FROM (SELECT MAX(g.timestamp) as maxTime, gu.userID as userID FROM GameUser gu INNER JOIN Game g ON g.gameID=gu.gameID GROUP BY gu.userID) temptable INNER JOIN User u on u.userID = temptable.userID where isRunning = 1 order by maxTime ASC limit 1");
+                }
+            } else {
+                $seedPlayer = $this->select("SELECT * FROM (SELECT * FROM User WHERE isRunning = 1 ORDER BY numGames ASC limit 15) orderedTable ORDER BY rand() LIMIT 1");
```

Comment:
If the plan is to have a significantly larger number of workers during the whole finals, it might be a good idea to up the limit in the orderedtable query for seeding (currently at 15). That was figured to be about half the number of workers, if the number of workers gets too larger in comparison there will be lots of duplicate seeding. On the other hand if the number set is too large in comparison to number of workers it will be too low of a chance for the players with the most need actually getting a game.